tests/api/test_flows.py

```diff
@@ -1308,6 +1308,54 @@ async def test_allows_for_same_time_if_event_attrs_are_different(
 
         assert len(set([fr.scheduled_start_time for fr in flow_runs])) == 5
 
+    @pytest.mark.parametrize(
+        "attrs",
+        [
+            [
+                dict(parameter_defaults=dict(x="a")),
+                dict(parameter_defaults=dict(x="b")),
+            ],
+            [dict(parameter_defaults=dict(x="a")), dict(parameter_defaults=None)],
+            [dict(parameter_defaults=dict(x="a")), dict(labels=["b"])],
+            [dict(labels=["c", "d"]), dict(labels=["c"])],
+            [dict(labels=None), dict(labels=["ef"])],
+            [
+                dict(labels=None),
+                dict(labels=[]),
+            ],  # the scheduler should distinguish between none vs. empty
+        ],
+    )
+    async def test_same_time_different_parameters_does_not_schedule_duplicates(
+        self, project_id, attrs
+    ):
+        now = pendulum.now("UTC")
+        clock1 = prefect.schedules.clocks.IntervalClock(
+            start_date=now.add(minutes=1),
+            interval=datetime.timedelta(minutes=2),
+            **attrs[0],
+        )
+        clock2 = prefect.schedules.clocks.IntervalClock(
+            start_date=now.add(minutes=1),
+            interval=datetime.timedelta(minutes=2),
+            **attrs[1],
+        )
+
+        flow = prefect.Flow(
+            name="Test Scheduled Flow",
+            schedule=prefect.schedules.Schedule(clocks=[clock1, clock2]),
+        )
+        flow.add_task(prefect.Parameter("x", default=1))
+        flow_id = await api.flows.create_flow(
+            project_id=project_id, serialized_flow=flow.serialize()
+        )
+        await models.FlowRun.where({"flow_id": {"_eq": flow_id}}).delete()
+        assert len(set((await api.flows.schedule_flow_runs(flow_id)))) == 10
+
+        await models.FlowRun.where({"flow_id": {"_eq": flow_id}}).update(
+            set=dict(idempotency_key=None)
```

Comment:
I gave this some thought and think we should:

- remove this piece from the test where we reset the idempotency key
- remove [this line](https://github.com/PrefectHQ/server/blob/master/src/prefect_server/api/runs.py#L53) so that idempotency keys are global and non-expiring; right the 24 hour window is arbitrary and users could easily control this when sending keys themselves by sending appropriately rounded timestamps